mirgecom/transport.py

```diff
@@ -0,0 +1,212 @@
+from mirgecom.eos import GasEOS
+
+
+@dataclass
+class TransportDependentVars:
+    """State-dependent quantities for :class:`TransportModel`.
+
+    Prefer individual methods for model use, use this
+    structure for visualization or probing.
+
+    .. attribute:: bulk_viscosity
+    .. attribute:: viscosity
+    .. attribute:: thermal_conductivity
+    .. attribute:: species_diffusivity
+    """
+
+    bulk_viscosity: np.ndarray
+    viscosity: np.ndarray
+    thermal_conductivity: np.ndarray
+    species_diffusivity: np.ndarray
+
+
+class TransportModel:
+    r"""Abstract interface to thermo-diffusive transport model class.
+
+    Transport model classes are responsible for
+    computing relations between fluid or gas state variables and
+    thermo-diffusive transport properties for those fluids.
+
+    .. automethod:: bulk_viscosity
+    .. automethod:: viscosity
+    .. automethod:: thermal_conductivity
+    .. automethod:: species_diffusivity
+    .. automethod:: viscosity2
+    """
+
+    def bulk_viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the bulk viscosity for the gas (${\mu}_{B}$)."""
+        raise NotImplementedError()
+
+    def viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the gas dynamic viscosity, $\mu$."""
+        raise NotImplementedError()
+
+    def viscosity2(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the 2nd coefficent of viscosity, $\lambda$."""
+        raise NotImplementedError()
+
+    def thermal_conductivity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the gas thermal_conductivity, $\kappa$."""
+        raise NotImplementedError()
+
+    def species_diffusivity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the vector of species diffusivities (${d}_{\alpha}$)."""
+        raise NotImplementedError()
+
+
+class SimpleTransport(TransportModel):
+    r"""Transport model with uniform, constant properties.
+
+    .. automethod:: __init__
+
+    Inherits from (and implements) :class:`TransportModel`.
+    """
+
+    def __init__(self, bulk_viscosity=0, viscosity=0,
+                 thermal_conductivity=0,
+                 species_diffusivity=None):
+        """Initialize uniform, constant transport properties."""
+        if species_diffusivity is None:
+            species_diffusivity = np.empty((0,), dtype=object)
+        self._mu_bulk = bulk_viscosity
+        self._mu = viscosity
+        self._kappa = thermal_conductivity
+        self._d_alpha = species_diffusivity
+
+    def bulk_viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the bulk viscosity for the gas, $\mu_{B}$."""
+        return self._mu_bulk
+
+    def viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the gas dynamic viscosity, $\mu$."""
+        return self._mu
+
+    def viscosity2(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the 2nd viscosity coefficent, $\lambda$."""
+        return self._mu_bulk - 2 * self._mu / 3
```

Comment:
Maybe document the math for this one similar to how you did for the methods in `PowerLawTransport`?